### PR TITLE
fix: Properly cache manifests for .ipa bundles containing multiple apps

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/20075

### CODE REVIEW

**Comment by KazuCocoa** [comment]
I haven't tested such a case, i'll take a look

File: lib/app-infos-cache.js [hunk 1]
```diff
@@ -22,130 +25,134 @@ export class AppInfosCache {
...
+   * @returns {Promise<import('@appium/types').StringRecord>} The payload of the manifest plist
+   */
+  async _putIpa(ipaPath) {
+    /** @type {import('@appium/types').StringRecord|undefined} */
+    let manifestPayload;
+    /** @type {Error|undefined} */
+    let lastError;
+    try {
+      await zip.readEntries(ipaPath, async ({entry, extractEntryTo}) => {
+        if (!IPA_ROOT_PLIST_PATH_PATTERN.test(entry.fileName)) {
```
Comment (KazuCocoa) [inline lib/app-infos-cache.js:101]: so, the main change is here?
The top of info plist (Payload/xxx.app/Info.plist) will be read. Watch app would be `Payload/xxx.app/Watch/yyy.app` so watch app will be ignored.

Can you leave a comment? I may forget the nested Watch app case here in the future...

(just a link I checked the structure)
https://medium.com/@programmingpassion/watchos-app-bundle-release-management-b6e58545b1c9

**Review by KazuCocoa — APPROVED** [review]
lg, roughly tested with existing ipa/app files I have

File: lib/app-infos-cache.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/app-infos-cache.js:102]: Added the link to the original issue

**Pull request merged**